pkg/filter/filter.go

```diff
@@ -192,12 +147,12 @@ func Predicate() predicate.Predicate {
 
 			// Required for the case when pods are deleted due to OS upgrade
 
-			if Owned(obj) && kernel.IsObjectAffine(obj) {
+			if f.owned(obj) && kernel.IsObjectAffine(obj) {
 				if e.ObjectOld.GetGeneration() == e.ObjectNew.GetGeneration() &&
 					e.ObjectOld.GetResourceVersion() == e.ObjectNew.GetResourceVersion() {
 					return false
 				} else {
-					log.Info(Mode+" Owned Generation or resourceVersion Changed for kernel affine object",
+					f.log.Info(f.mode+" Owned Generation or resourceVersion Changed for kernel affine object",
 						"Name", obj.GetName(), "Type", reflect.TypeOf(obj).String())
 					if reflect.TypeOf(obj).String() == "*v1.DaemonSet" && e.ObjectOld.GetGeneration() != e.ObjectNew.GetGeneration() {
 						err := lifecycle.New(clients.Interface).UpdateDaemonSetPods(obj)
```

Comment:
Could we store a `clients.Interface` in the `filter` struct?

Reply:
I used `lifecycle` and added UTs for Update predicate.